stdlib/LinearAlgebra/test/lapack.jl

```diff
@@ -410,7 +410,7 @@ end
         @test_throws DimensionMismatch LAPACK.stev!('U',d,rand(elty,10))
         @test_throws DimensionMismatch LAPACK.stebz!('A','B',zero(elty),zero(elty),0,0,-1.,d,rand(elty,10))
         @test_throws DimensionMismatch LAPACK.stegr!('N','A',d,rand(elty,11),zero(elty),zero(elty),0,0)
-        @test_throws DimensionMismatch LAPACK.stein!(d,zeros(elty,10),zeros(elty,10),zeros(BlasInt,10),zeros(BlasInt,10))
+        @test_throws DimensionMismatch LAPACK.stein!(d,zeros(elty,11),zeros(elty,10),zeros(BlasInt,10),zeros(BlasInt,10))
```

Comment:
It would be good to also have positive tests, so that this functionality cannot regress in the future. Like have an example where both `dv` and `ev` have the same length, and then call `eigen`, `eigvals` etc. on them. For simplicity, you could choose `ev = zeros(eltype(dv), length(dv))`, then you have the eigenvalues on the diagonal, and you can check the numerical values easily.